en: add doc about volume snapshot backup/restore across multiple kubernetes (#2406) (#2446)
pingcap/docs-tidb-operator: en/backup-restore-by-ebs-snapshot-faq.md

**Solution:** This issue might occur if you have disabled the feature of "resolved ts" in TiKV or PD. Check the configuration of TiKV and PD:

- For TiKV, confirm if you set `resolved-ts.enable = false` or `raftstore.report-min-resolved-ts-interval = "0s"`. If so, remove these configurations.
- For PD, confirm if you set `pd-server.min-resolved-ts-persistence-interval = "0s"`. If so, remove this configuration.

## Backup failed due to execution twice

**Issue:** [#5143](https://github.com/pingcap/tidb-operator/issues/5143)

**Symptom:** You get the error that contains `backup meta file exists`, and the backup pod is scheduled twice.

**Solution:** This issue might occur if the first backup pod is evicted by Kubernetes due to node resource pressure. You can configure `PriorityClass` and `ResourceRequirements` to reduce the possibility of eviction. For more details, refer to the [comment of issue](https://github.com/pingcap/tidb-operator/issues/5143#issuecomment-1654916830).

## Save time for backup by controlling snapshot size calculation level

**Symptom:** Scheduled backup can't be completed in the expected window due to the cost of snapshot size calculation.

> **Solution:** By default, both full size and incremental size are calculated by calling the AWS service, which might take several minutes. You can set `spec.template.calcSizeLevel` to `full` to skip incremental size calculation, set it to `incremental` to skip full size calculation, and set it to `none` to skip both calculations.

Reviewer: ```suggestion
**Solution:** By default, both full size and incremental size are calculated by calling the AWS service, which might take several minutes. You can set `spec.template.calcSizeLevel` to `full` to skip incremental size calculation, set it to `incremental` to skip full size calculation, and set it to `none` to skip both calculations.

## How to configure the TTL for the backup init job

Backup init job will do backup preparation work including pause GC, some PD schedulers and also suspend lightning.  Default TTL of 10 minutes is associated with the init job in case it could stuck.  You can change the TTL by setting `spec.template.volumeBackupInitJobMaxActiveSeconds` attribute of spec of volumebackup.
```

## Save time for backup by controlling snapshot size calculation level

**Symptom:** Scheduled backup can't be completed in the expected window due to the cost of snapshot size calculation.

**Solution:** By default, both full size and incremental size are calculated by calling the AWS service, which might take several minutes. You can set `spec.template.calcSizeLevel` to `full` to skip incremental size calculation, set it to `incremental` to skip full size calculation, and set it to `none` to skip both calculations.

## How to configure the TTL for the backup init job

The backup init job will handle backup preparations, including pausing GC, certain PD schedulers, and suspending Lightning. By default, a TTL of 10 minutes is associated with the init job in case it gets stuck. You can change the TTL by setting the `spec.template.volumeBackupInitJobMaxActiveSeconds` attribute of spec of volumebackup.

## How to flow control to snapshots deletion

> EBS snapshot backup GC is performed on one volumebackup at a time. For larger clusters with EBS snapshot backups, there might still be a significant number of snapshots for a single volume backup. Therefore, flow control is necessary for snapshot deletion. You can manage the expected ration in a single data plane by setting the `spec.template.snapshotsDeleteRatio` parameter of the backup schedule CRD. The default value is 1.0, which ensures no more than one snapshot deletion per second.

Reviewer: ```suggestion
EBS snapshot backup GC is performed on one volumebackup at a time. For larger clusters with EBS snapshot backups, there might still be a significant number of snapshots for a single volume backup. Therefore, flow control is necessary for snapshot deletion. You can manage the expected ratio in a single data plane by setting the `spec.template.snapshotsDeleteRatio` parameter of the backup schedule CRD. The default value is 1.0, which ensures no more than one snapshot deletion per second.
```

**Symptom:** Scheduled backup can't be completed in the expected window due to the cost of snapshot size calculation.

**Solution:** By default, both full size and incremental size are calculated by calling the AWS service, which might take several minutes. You can set `spec.template.calcSizeLevel` to `full` to skip incremental size calculation, set it to `incremental` to skip full size calculation, and set it to `none` to skip both calculations.

## How to configure the TTL for the backup init job

> Backup init job will do backup preparation work including pause GC, some PD schedulers and also suspend lightning.  Default TTL of 10 minutes is associated with the init job in case it could stuck.  You can change the TTL by setting `spec.template.volumeBackupInitJobMaxActiveSeconds` attribute of spec of volumebackup.

Reviewer 2: ```suggestion
The backup init job will handle backup preparations, including pausing GC, certain PD schedulers, and suspending Lightning. By default, a TTL of 10 minutes is associated with the init job in case it gets stuck. You can change the TTL by setting the `spec.template.volumeBackupInitJobMaxActiveSeconds` attribute of spec of volumebackup.
```

## How to flow control to snapshots deletion

> EBS snapshot backup GC is done one volumebackup each time. For larger clusters EBS snapshot backup, number of snapshots for a single volumebackup could still be very large, so, flow control to the snapshot deletion is still needed.  You can control expected ration in a single data plane by setting the paramter `spec.template.snapshotsDeleteRatio` of the backup schedule CRD. Its default value is 1.0, which means no more than 1 snapshot deletion in a second.

Reviewer 2: ```suggestion
EBS snapshot backup GC is performed on one volumebackup at a time. For larger clusters with EBS snapshot backups, there might still be a significant number of snapshots for a single volume backup. Therefore, flow control is necessary for snapshot deletion. You can manage the expected ration in a single data plane by setting the `spec.template.snapshotsDeleteRatio` parameter of the backup schedule CRD. The default value is 1.0, which ensures no more than one snapshot deletion per second.
```
